src/util/redisConnector.js

```diff
@@ -81,23 +98,38 @@ const RedisDB = {
   /**
    * Used to set value in redis depending on the key and the value type
    * @param {*} key key for which value needs to be stored
-   * @param {*} value value to be stored in redis
-   * @param {*} isValJson set to false if value is not a json object
+   * @param {*} value to be stored in redis send it in array format [field1, value1, field2, value2]
+   *                  if Value is an object
+   * @param {*} expiryTime expiry time of data in redis by default 1 hr
+   * @param {*} isValJson set to false if value is not a json object 
+   * 
    */
-
-  async setVal(key, value, isValJson = true) {
-    const dataExpiry = 60 * 60; // 1 hour
+  async setVal(key, value, expiryTime = 60 * 60) {
     try {
       await this.checkAndConnectConnection(); // check if redis is connected and if not, connect
-      const valueToStore = isValJson ? JSON.stringify(value) : value;
-      const bytes = Buffer.byteLength(valueToStore, "utf-8");
-      await this.client.setex(key, dataExpiry, valueToStore);
+      let bytes;
+      if (typeof value === "object") {
+        const valueToStore = value.map(element => {
+          if (typeof element === "object") {
+            return JSON.stringify(element);
+          }
+          return element;
+        })
+        bytes = Buffer.byteLength(JSON.stringify(value), "utf-8");
+        await this.client.multi()
+          .hmset(key, ...valueToStore)
+          .expire(key, expiryTime)
+          .exec();
+      } else {
+        bytes = Buffer.byteLength(value, "utf-8");
+        await this.client.setex(key, expiryTime, value);
+      }
```

Comment:
Remove timestamp as a stats label for all occurrences

Reply:
Done in https://github.com/rudderlabs/rudder-transformer/pull/2107